### PR TITLE
Verify that users don't enable DTC on top of a database configured to use Voron

### PR DESCRIPTION
**Pull request by danielmarbach**
Decided to throw.

@Particular/engineers please review

### CODE REVIEW

**Comment by synhershko** [comment]
I'd throw if Voron is used, period :)

**Comment by SimonCropp** [comment]
so what about people who are aware that voron doesnt support DTC but still want to use DTC for other things? doesnt this block a valid use case?

**Comment by danielmarbach** [comment]
Change it to log.Warn?

Am Montag, 27. April 2015 schrieb Simon Cropp :

> so what about people who are aware that voron doesnt support DTC but still
> want to use DTC for other things? doesnt this block a valid use case?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus.RavenDB/pull/68#issuecomment-96614379>
> .
>


-- 

Daniel Marbach

Solution Architect, Particular Software

Email:  daniel.marbach@particular.net

Web:    www.particular.net
<http://t.signauxneuf.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJN7t5XX4RzDjlVS9QB88pTc_dW3Ljg7x56dWyBdgBXs402?t=http%3A%2F%2Fwww.particular.net%2F&si=4896787407044608&pi=33cc6202-1f31-4040-8d6c-e70641f44bba>

**Comment by andreasohlund** [comment]
Sounds resonable, should we provide a override?

**Comment by danielmarbach** [comment]
@andreasohlund what do you mean with override?

**Comment by andreasohlund** [comment]
A way to say, "I know what I'm doing, please let me use Voron"

**Comment by danielmarbach** [comment]
@andreasohlund see update. Good?

**Comment by andreasohlund** [comment]
Looks good!

**Comment by danielmarbach** [comment]
@SimonCropp I need the simon cropp spell checker mechanical turk ;)

File: src/NServiceBus.RavenDB.Tests/Persistence/DocumentStoreExtensionsForVoron.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        public static IDisposable SetupVoronTest(this DocumentStore store)
+        {
+            store.Initialize();
+            store.DatabaseCommands.GlobalAdmin.CreateDatabase(new DatabaseDocument
+            {
+                Id = "VoronTest",
+                Settings = new Dictionary<string, string>
+                    {
+                        { "Raven/StorageTypeName", "voron" },
+                        { "Raven/DataDir", "~/VoronTest" }
```
Comment (SimonCropp) [inline src/NServiceBus.RavenDB.Tests/Persistence/DocumentStoreExtensionsForVoron.cs]: Test should be able to run in parallel. Perhaps use a temp directory?

Comment (danielmarbach) [inline src/NServiceBus.RavenDB.Tests/Persistence/DocumentStoreExtensionsForVoron.cs]: I'll change that. Unrelated question: Do I need to update wire compat tests
or anything else as well somewhere before releasing?

Am Montag, 4. Mai 2015 schrieb Simon Cropp :

> In
> src/NServiceBus.RavenDB.Tests/Persistence/DocumentStoreExtensionsForVoron.cs
> <https://github.com/Particular/NServiceBus.RavenDB/pull/68#discussion_r29625639>
> :
>
> > +    using System.Collections.Generic;
> > +    using Raven.Abstractions.Data;
> > +    using Raven.Client.Document;
> > +
> > +    internal static class DocumentStoreExtensionsForVoron
> > +    {
> > +        public static IDisposable SetupVoronTest(this DocumentStore store)
> > +        {
> > +            store.Initialize();
> > +            store.DatabaseCommands.GlobalAdmin.CreateDatabase(new DatabaseDocument
> > +            {
> > +                Id = "VoronTest",
> > +                Settings = new Dictionary<string, string>
> > +                    {
> > +                        { "Raven/StorageTypeName", "voron" },
> > +                        { "Raven/DataDir", "~/VoronTest" }
>
> Test should be able to run in parallel. Perhaps use a temp directory?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus.RavenDB/pull/68/files#r29625639>
> .
>


-- 

Daniel Marbach

Solution Architect, Particular Software

Email:  daniel.marbach@particular.net

Web:    www.particular.net
<http://t.signauxneuf.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJN7t5XX4RzDjlVS9QB88pTc_dW3Ljg7x56dWyBdgBXs402?t=http%3A%2F%2Fwww.particular.net%2F&si=4896787407044608&pi=33cc6202-1f31-4040-8d6c-e70641f44bba>

Comment (andreasohlund) [inline src/NServiceBus.RavenDB.Tests/Persistence/DocumentStoreExtensionsForVoron.cs]: Yes there is a test Yak in need of your capable shaving :)

http://builds.particular.net/viewType.html?buildTypeId=ProductionStagingTests_RavenDBPersistenceCompatibility

**Comment by danielmarbach** [comment]
For me this one is good to go. Any objections?

**Comment by andreasohlund** [comment]
+1

**Comment by danielmarbach** [comment]
I plan to shave the yak after the release

Am Dienstag, 5. Mai 2015 schrieb Andreas Öhlund :

> +1
>
> On Tue, May 5, 2015 at 5:19 PM, Daniel Marbach <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>>
> wrote:
>
> > For me this one is good to go. Any objections?
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/Particular/NServiceBus.RavenDB/pull/68#issuecomment-99112884
> >
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus.RavenDB/pull/68#issuecomment-99155592>
> .
>


-- 

Daniel Marbach

Solution Architect, Particular Software

Email:  daniel.marbach@particular.net

Web:    www.particular.net
<http://t.signauxneuf.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJN7t5XX4RzDjlVS9QB88pTc_dW3Ljg7x56dWyBdgBXs402?t=http%3A%2F%2Fwww.particular.net%2F&si=4896787407044608&pi=33cc6202-1f31-4040-8d6c-e70641f44bba>

**Pull request merged**